pkg/webui/console/containers/lora-cloud-gls-form/index.js

```diff
@@ -0,0 +1,166 @@
+import { isNotFoundError } from '@ttn-lw/lib/errors/utils'
+import attachPromise from '@ttn-lw/lib/store/actions/attach-promise'
+
+import {
+  setAppPkgDefaultAssoc,
+  getAppPkgDefaultAssoc,
+  deleteAppPkgDefaultAssoc,
+} from '@console/store/actions/application-packages'
+
+import { selectSelectedApplicationId } from '@console/store/selectors/applications'
+import {
+  selectApplicationPackageDefaultAssociation,
+  selectGetApplicationPackagesError,
+} from '@console/store/selectors/application-packages'
+
+const m = defineMessages({
+  tokenDescription: 'Geolocation access token as configured within LoRa Cloud',
+  setLoRaCloudToken: 'Set LoRa Cloud token',
+  deleteWarning:
+    'Are you sure you want to delete the LoRaCloud Geolocation token? This action cannot be undone.',
+})
+
+const validationSchema = Yup.object()
+  .shape({
+    data: Yup.object().shape({
+      token: Yup.string().required(sharedMessages.validateRequired),
+      query: Yup.string().oneOf(['TOARSSI']),
+    }),
+  })
+  .noUnknown()
+
+const defaultValues = {
+  data: {
+    token: '',
+    query: 'TOARSSI',
+  },
+}
+
+const promisifiedSetAppPkgDefaultAssoc = attachPromise(setAppPkgDefaultAssoc)
+const promisifiedDeleteAppPkgDefaultAssoc = attachPromise(deleteAppPkgDefaultAssoc)
+
+const LoRaCloudGLSForm = () => {
+  const [error, setError] = useState('')
+  const appId = useSelector(selectSelectedApplicationId)
+  const selector = ['data']
+
+  const dispatch = useDispatch()
+  const defaultAssociation = useSelector(state =>
+    selectApplicationPackageDefaultAssociation(state, LORA_CLOUD_GLS.DEFAULT_PORT),
+  )
+  const packageError = useSelector(selectGetApplicationPackagesError)
+  const initialValues = validationSchema.cast(defaultAssociation || defaultValues)
+
+  const handleSubmit = useCallback(
+    async values => {
+      try {
+        await dispatch(
+          promisifiedSetAppPkgDefaultAssoc(appId, LORA_CLOUD_GLS.DEFAULT_PORT, {
+            package_name: LORA_CLOUD_GLS.DEFAULT_PACKAGE_NAME,
+            ...values,
+          }),
+        )
+        toast({
+          title: 'LoRa Cloud',
+          message: sharedMessages.tokenUpdated,
+          type: toast.types.SUCCESS,
+        })
+      } catch (error) {
+        setError(error)
+      }
+    },
+    [appId, dispatch],
+  )
+
+  const handleDelete = useCallback(async () => {
+    try {
+      await dispatch(
+        promisifiedDeleteAppPkgDefaultAssoc(appId, LORA_CLOUD_GLS.DEFAULT_PORT, {
+          package_name: LORA_CLOUD_GLS.DEFAULT_PACKAGE_NAME,
+        }),
+      )
+      toast({
+        title: 'LoRa Cloud',
+        message: sharedMessages.tokenDeleted,
+        type: toast.types.SUCCESS,
+      })
+    } catch (error) {
+      setError(error)
+    }
+  }, [appId, dispatch])
+
+  if (packageError && !isNotFoundError(packageError)) {
+    throw error
+  }
+
+  return (
+    <RequireRequest
+      requestAction={getAppPkgDefaultAssoc(appId, LORA_CLOUD_GLS.DEFAULT_PORT, selector)}
```

Comment:
Mb for later PRs, but I would really prefer having this fetched once the user lands on `/console/applications/{app_id}/integrations/lora-cloud`. Currently, once you expand gls/das form there is a slight delay before the form is rendered before fetching. What is more, you get this delay every time you collapse and expand the section.